db/background_mgr.go

```diff
@@ -52,47 +76,160 @@ type BackgroundManagerStatus struct {
 // BackgroundManagerProcessI is an interface satisfied by any of the background processes
 // Examples of this: ReSync, Compaction
 type BackgroundManagerProcessI interface {
+	Init(clusterStatus []byte) error
 	Run(options map[string]interface{}, terminator chan struct{}) error
 	GetProcessStatus(status BackgroundManagerStatus) ([]byte, error)
 	ResetStatus()
 }
 
 func (b *BackgroundManager) Start(options map[string]interface{}) error {
+	b.terminator = make(chan struct{}, 1)
```

Comment:
Start seems to be called based on a db-scoped BackgroundManager instance in some cases (db.AttachmentCompactionManager, db.ResyncManager).  How are we guarding against it being called more than once?

It looks like db.TombstoneCompactionManager handles that externally (with db.CompactState), but it would make more sense for the BackgroundManagers to handle, I think?  BackgroundProcessState seems like it might be a reasonable way to manage that.

Reply:
TombstoneCompactionManager does handle it externally but that can be removed now really. The markStart() function handles this. Although granted it doesn't look like its super safe. Could probably switch that out a little for some sort of compare and swap deal.